paasta_tools/setup_marathon_job.py

```diff
@@ -334,7 +335,8 @@ def log_bounce_action(line: str, level: str='debug') -> None:
         if task in tasks_to_kill:
             hostname = task.host
             try:
-                reserve_all_resources([hostname])
+                if enable_maintenance_reservation:
```

Comment:
It looks like the whole purpose of this loop is to reserve. It would be the slightest bit faster if you put the if block before the loop at all

Reply:
👍